### PR TITLE
Feature/cover v2 general cover edit

### PR DESCRIPTION
**Pull request by danoctavian**

### CODE REVIEW

File: contracts/modules/cover/Cover.sol [hunk 1]
```diff
@@ -415,119 +268,46 @@ contract Cover is ICover, MasterAwareV2 {
...
+      uint refundInPaymentAsset =
+      refundInCoverAsset * 1e18 / pool().getTokenPrice(cover.payoutAsset)
+      * (pool().getTokenPrice(buyCoverParams.payoutAsset) / 10 ** paymentAssetDecimals);
 
-      // write the new staking pool with modified parameters
-      coverChunksForCover[newCoverId].push(newCoverChunks[i]);
+      if (refundInPaymentAsset < premiumInPaymentAsset) {
+        // retrieve extra required payment
+        retrievePayment(premiumInPaymentAsset - refundInPaymentAsset, buyCoverParams.paymentAsset);
+      }
```
Comment (horodnicdragos) [inline contracts/modules/cover/Cover.sol:307]: Should there be a 10% NXM mint for the paid difference (premiumInPaymentAsset - refundInPaymentAsset) just like in buyCover?

File: contracts/modules/cover/Cover.sol [hunk 2]
```diff
@@ -153,6 +154,19 @@ contract Cover is ICover, MasterAwareV2 {
       "Cover: Payout asset is not supported"
     );
 
+    (uint coverId, uint premiumInPaymentAsset, uint totalPremiumInNXM) = _buyCover(params, coverChunkRequests);
+    require(premiumInPaymentAsset <= params.maxPremiumInAsset, "Cover: Price exceeds maxPremiumInAsset");
+    retrievePayment(premiumInPaymentAsset, params.payoutAsset);
```
Comment (horodnicdragos) [inline contracts/modules/cover/Cover.sol:159]: What about paying with NXM?

File: contracts/modules/cover/Cover.sol [hunk 3]
```diff
@@ -234,179 +243,23 @@ contract Cover is ICover, MasterAwareV2 {
...
-    );
-
-    // TODO: add check for max 20% MCR check for activeCoverAmountInNXM
-
-    CoverChunk[] storage originalCoverChunks = coverChunksForCover[params.coverId];
-
-    // reduce period
+    uint totalPreviousCoverAmountInNXM = 0;
+    // rollback previous cover
     for (uint i = 0; i < originalCoverChunks.length; i++) {
```
Comment (horodnicdragos) [inline contracts/modules/cover/Cover.sol:263]: Store originalCoverChunks.length in memory before the loop so yout don't read from storage at every iteration.

File: contracts/modules/cover/Cover.sol (hunk 1 above)
Comment (danoctavian) [reply contracts/modules/cover/Cover.sol:307]: conifmed yes we want this

File: contracts/modules/cover/Cover.sol [hunk 4]
```diff
@@ -156,11 +156,16 @@ contract Cover is ICover, MasterAwareV2 {
 
     (uint coverId, uint premiumInPaymentAsset, uint totalPremiumInNXM) = _buyCover(params, coverChunkRequests);
     require(premiumInPaymentAsset <= params.maxPremiumInAsset, "Cover: Price exceeds maxPremiumInAsset");
-    retrievePayment(premiumInPaymentAsset, params.payoutAsset);
 
-    // mint 10% of paid premium value to user
-    tokenController().mint(params.owner, totalPremiumInNXM / 10);
+    if (params.payWithNXM) {
+      tokenController().burnFrom(msg.sender, totalPremiumInNXM);
```
Comment (horodnicdragos) [inline contracts/modules/cover/Cover.sol:161]: This should be 90%, shouldn't it?

Comment (danoctavian) [reply contracts/modules/cover/Cover.sol:161]: we're not doing that anymore

**Pull request merged**